core/settings_dialog.py

```
@@ -289,6 +323,114 @@ def setup_ui(self):
+        if len(parts) < 2:
+            QMessageBox.warning(self, "Install Plugin", f"Could not parse repo from: {repo_input}")
+            return
+        owner, repo = parts[0], parts[1]
+
+        # Try main then master branch
+        installed = False
+        last_error = None
+        for branch in ('main', 'master'):
+            zip_url = f"https://github.com/{owner}/{repo}/archive/refs/heads/{branch}.zip"
+            try:
+                with urllib.request.urlopen(zip_url, timeout=30) as resp:
```

Ruff (0.15.9)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)